backend/utils/graph_db_utils_test.py

```diff
@@ -0,0 +1,64 @@
+from unittest.mock import MagicMock
+from neo4j import Driver, Session
+import pytest
+# We assign an alias to "test_connection" to avoid pytest treating it as another test function
+from utils import test_connection as verify_connection, create_goal
+
+@pytest.fixture
+def mock_session():
+    return MagicMock(spec=Session)
+
+
+@pytest.fixture
+def mock_driver(mocker, mock_session):
+    mock_driver = mocker.patch("utils.graph_db_utils.driver", return_value=MagicMock(spec=Driver))
+    mock_driver.session.return_value = mock_session
+    return mock_driver
+
+
+def remove_whitespace_and_newlines(original):
+    return " ".join(original.replace(r'\n', ' ').replace(r'\r', '').split())
+
+
+def test_database_connectivity_is_healthy(mock_driver):
+    mock_driver.verify_connectivity.return_value = None
+
+    connected = verify_connection()
```

Comment:
Not sure if it's more readable but technically we could shorten these tests like so
```suggestion
def test_database_connectivity_is_healthy(mock_driver):
    mock_driver.verify_connectivity.return_value = None

    assert verify_connection()
    mock_driver.verify_connectivity.assert_called_once()
```

Again, your pick, I think the current version is more readable, just creates a variable that is used once

Reply:
It's a good point.

In my case, I'm attempting to follow the "arrange, act, assert" approach with a clear separation between each stage. For that reason, I've assigned to a temporary variable in the "act" stage to separate this from the "assert" stage.

Your point is very valid. I've chosen to keep these separate for consistency with the other tests but I've no objections to changing this.